WordPress/src/main/java/org/wordpress/android/ui/uploads/UploadService.java

```diff
@@ -107,9 +112,14 @@ public void onDestroy() {
             mPostUploadHandler.unregister();
         }
 
-        // Update posts with any completed uploads in our post->media map
-        for (Integer postId : sCompletedMediaByPost.keySet()) {
+        // Update posts with any completed AND failed uploads in our post->media map
+        ConcurrentHashMap<Integer, List<MediaModel>> tmpMap = new ConcurrentHashMap<>();
```

Comment:
IIUC it is true it would be okay to just use a plain Map here in `onDestroy`, but I wanted to keep the same code as in `stopServiceIfUploadsComplete()`, where the map needs to be concurrent-aware as per the several entry points. As a matter of fact, given these were identical pieces of code, refactored them into one method in 7418804